tests/test_askrene.py

```diff
@@ -0,0 +1,414 @@
+from fixtures import *  # noqa: F401,F403
+from pyln.client import RpcError
+from utils import (
+    only_one, first_scid, GenChannel, generate_gossip_store,
+    TEST_NETWORK
+)
+import os
+import pytest
+import time
+import shutil
+
+
+def test_layers(node_factory):
+    """Test manipulating information in layers"""
+    l1, l2, l3 = node_factory.line_graph(3, wait_for_announce=True)
+
+    assert l2.rpc.askrene_listlayers() == {'layers': []}
+    assert l2.rpc.askrene_listlayers('test_layers') == {'layers': []}
+
+    expect = {'layer': 'test_layers',
+              'disabled_nodes': [],
+              'created_channels': [],
+              'constraints': []}
+    l2.rpc.askrene_disable_node('test_layers', l1.info['id'])
+    expect['disabled_nodes'].append(l1.info['id'])
+    assert l2.rpc.askrene_listlayers('test_layers') == {'layers': [expect]}
+    assert l2.rpc.askrene_listlayers() == {'layers': [expect]}
+    assert l2.rpc.askrene_listlayers('test_layers2') == {'layers': []}
+
+    # Tell it l3 connects to l1!
+    l2.rpc.askrene_create_channel('test_layers',
```

Comment:
Isn't this going to fail 50% of the times? I am not sure how the node ids are created in `pyln-testing` but if they are random then there is a 50% chance that `l1.info['id'] > l3.info['id']` and create channel fails.

Reply:
Yeah, I cheated.  Node ids are deterministic, so this works, even though I really should dynamically determine direction.